api_v3/lib/types/resource/KalturaEntryResource.php

```diff
@@ -36,7 +36,41 @@ public function validateForUsage($sourceObject, $propertiesToSkip = array())
 			$srcEntry->getMediaType() == KalturaMediaType::LIVE_STREAM_FLASH ||
 				myEntryUtils::isLiveClippingEntry($srcEntry))
 			return;
+
+		if (!$this->checkIfFileExist())
+			throw new KalturaAPIException(KalturaErrors::FILE_DOESNT_EXIST);
+	}
+	
+	/* (non-PHPdoc)
+	 * @see KalturaResource::validateEntry()
+	 */
+	public function validateEntry(entry $dbEntry, $validateLocal = false)
+	{
+		parent::validateEntry($dbEntry, $validateLocal);
 		
+		$srcEntry = entryPeer::retrieveByPK($this->entryId);
+		if(!$srcEntry)
+			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $this->entryId);
+		if ($validateLocal && !$this->checkIfFileExist(true))
```

Comment:
Since you are in API , no sense in raising kCoreException.
try - throw new KalturaAPIException(APIErrors::SOURCE_FILE_REMOTE); ......The source file is remote, no local file found.....